ipatests/test_integration/test_installation.py

```diff
@@ -1108,6 +1115,79 @@ def run_request(url, expected_stdout=None, expected_stderr=None):
                 expected_stdout=f'href="https://{self.master.hostname}/'
             )
 
+    @master_unistalled
+    def test_hostname_parameter(self):
+        """
+        Test for #2692 ipa-server-install ignores --hostname
+
+        https://pagure.io/freeipa/issue/2692
+        """
+        original_hostname = self.master.hostname
+        new_hostname = "new." + original_hostname
+        ip_address = self.master.ip
+        hosts = self.master.get_file_contents('/etc/hosts')
+        new_hosts = str(hosts) + "\n{}        {}".format(
+            ip_address, new_hostname)
+        self.master.put_file_contents('/etc/hosts', new_hosts)
+        result = tasks.install_master(self.master,
+                                      setup_dns=False,
+                                      extra_args=['--hostname',
+                                                  new_hostname])
+        try:
+            assert result.returncode == 0
```

Comment:
assert is redundant  since `raiseonerr` parameter of `tasks.install_master` is True by default